packages/shopify-cart/src/utils/transformShopifyLineToNacelleLine.ts

```diff
@@ -1,15 +1,28 @@
 import { CartLine, NacelleCartLine } from '../types/cart.type';
 
-export default function (shopifyLines: CartLine[]): NacelleCartLine[] {
-  return shopifyLines?.map((shopifyLine) => {
-    const transformedLine: CartLine = { ...shopifyLine };
-    const nacelleEntryId = transformedLine.nacelleEntryId?.value as string;
-    delete transformedLine.nacelleEntryId;
+export interface TransformShopifyCartLineParams {
+  lines: CartLine[];
+  shopifyShopId: string;
+  locale: string;
+}
+
+export default function ({
+  lines,
+  shopifyShopId,
+  locale
+}: TransformShopifyCartLineParams): NacelleCartLine[] {
+  return lines?.map((line) => {
+    const shopifyVariantId = line.merchandise.sourceEntryId
+      .split('gid://shopify/ProductVariant/')[1]
+      .replace('/', '');
+
     return {
-      ...transformedLine,
+      ...line,
       merchandise: {
-        ...transformedLine.merchandise,
-        nacelleEntryId
+        ...line.merchandise,
+        nacelleEntryId: btoa(
+          `id://SHOPIFY/${shopifyShopId}/default/PRODUCT_VARIANT/${shopifyVariantId}/${locale}`
+        )
```

Comment:
Did we want to handle Node differently? I think we determined that the btoa/atob methods should be sufficient for our use case, but wanted to verify.

Reply:
I wish we could, but unfortunately we can't rely on Node's `atob`, for a couple of reasons:
1. It was introduced in Node `16.0.0`, and we have customers that build their projects with Node 14.
2. For reasons that aren't clear to me, `atob` and `btoa` are listed as [Stability 3: Legacy](https://nodejs.org/docs/latest-v18.x/api/all.html#all_documentation_stability-index). Both Node 16 and Node 18 docs suggest that people use `Buffer.from(data, 'base64')` instead.

![Node 16 atob stability 3 legacy](https://user-images.githubusercontent.com/5732000/192373355-c9119d9c-76fd-4493-a5a1-f02ce91edd75.png)

![Node 18 atob btoa legacy](https://user-images.githubusercontent.com/5732000/192373373-081f94da-cbeb-4ce1-8508-7b5e5e360b15.png)

Reply:
So @alexcasche we should implement an isomorphic base64 decoding. For example: https://github.com/getnacelle/nacelle-js/blob/ENG-6843/shopify-cart-1.0.0/packages/shopify-cart/src/utils/getShopifyIdFromNacelleId.ts#L8-L12

Since we'd be doing that in multiple places, it might make sense to have a single utility function that we can use wherever decoding needs to happen.

Reply:
Worth calling out that that line is decoding a base64 encoded gid and the one here encoding one. This might be the only place we're base64 encoding anything, since all the shopify stuff is decoded by default.

Reply:
Totally. I edited the comment above to remove the suggestion to pull it out into a utility function.